include/matx/operators/isclose.h

```diff
@@ -0,0 +1,109 @@
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/scalar_ops.h"
+#include "matx/operators/base_operator.h"
+
+namespace matx
+{
+
+  namespace detail {
+    template <typename Op1, typename Op2> 
+    class IsCloseOp : public BaseOp<IsCloseOp<Op1, Op2>>
+    {
+      public:
+        using matxop = bool;
+        using scalar_type = typename remove_cvref_t<Op2>::scalar_type;
+        using inner_type = typename inner_op_type_t<scalar_type>::type;
+
+        __MATX_INLINE__ std::string str() const { return "isclose()"; }
+
+        __MATX_INLINE__ IsCloseOp(Op1 op1, Op2 op2, double rtol, double atol) : 
+          op1_(op1), op2_(op2), rtol_(static_cast<inner_type>(rtol)), atol_(static_cast<inner_type>(atol)) 
+        {
+          static_assert(op1.Rank() == op2.Rank(), "Operator ranks must match in isclose()");
+          for (int32_t i = 0; i < op2.Rank(); i++) {
+            MATX_ASSERT_STR(op1.Size(i) == op2.Size(i), matxInvalidDim, 
+                "Size of each dimension must match in isclose()");
+          }
+        }
+
+        template <typename... Is>
+          __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ int operator()([[maybe_unused]] Is... indices) const 
+          {
+
+            return static_cast<int>(detail::_internal_abs(op1_(indices...) - op2_(indices...)) <= 
+               static_cast<inner_type>(atol_) + static_cast<inner_type>(rtol_) * detail::_internal_abs(op2_(indices...)));
+          }
+
+        static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank()
+        {
+          return remove_cvref_t<Op1>::Rank();
+        }
+
+        constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t Size(int dim) const
+        {
+          return op1_.Size(dim);
```

Comment:
In Binary ops we do size like the following (should we match?):

      constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ auto Size(int dim) const noexcept
      {
        index_t size1 = detail::get_expanded_size<Rank()>(op1_, dim);
        index_t size2 = detail::get_expanded_size<Rank()>(op2_, dim);
        return detail::matx_max(size1,size2);
      }